tests/linearizability/client.go

```diff
@@ -89,3 +89,26 @@ func (c *recordingClient) Put(ctx context.Context, key, value string) error {
 	})
 	return nil
 }
+
+func (c *recordingClient) Delete(ctx context.Context, key string) error {
+	callTime := time.Now()
+	resp, err := c.client.Delete(ctx, key)
+	returnTime := time.Now()
+	if err != nil {
+		return err
```

Comment:
Even on error you should record an Delete operation. Reason is if the error was connection issue the delete may still be persisted in etcd.